File: BatchDownloader/TwitchAPI/APIData.vb
''Data Types sent out by API for the downloader to use
Public Class VideoList
    Public _total As Integer
    Public Videos As IList(Of VideoData)
    Public _links As Dictionary(Of String, String)
End Class

Public Class VideoData
    Public title As String
    Public channel As Dictionary(Of String, String) ' Name and display name
    Public recorded_at As String
    Public broadcast_id As String 'what is this?
    Public _id As String
    Public _links As Dictionary(Of String, String)
    Public embed As String
    Public url As String
    Public views As Integer
    Public preview As String
    Public length As Long
    Public game As String
    Public description As String
End Class
''JSON error
Public Class APIError
    Public message As String
    Public status As Integer
    'Public error as string
End Class
''Old FLV vods
Public Class VideoFLVStreamInfo
    Public api_id As String
    Public start_offset As Long
    Public end_offset As Long
    Public play_offset As Long
    Public increment_view_count_url As String
    Public path As String
    Public duration As Long
    Public broadcaster_software As String
    Public channel As String
    Public chunks As Dictionary(Of String, List(Of VideoFLVPart))
    'Public restrictions As List(Of String)
    Public preview_small As String
    Public preview As String
    Public vod_ad_frequency As Integer
    Public vod_ad_length As Integer
    Public redirect_api_id As String 'Redirects for v urls when this is an b/c link
    Public muted_segments As String() 'is this used
    Public can_highlight As Boolean
End Class

Public Class VideoFLVPart
    Public url As String
    Public vod_count_url As String
    Public length As Long
    Public upkeep As String
End Class

''AccessToken for new TS Vods
Public Class AccessToken
    Public token As String
    Public sig As String
End Class

File: TwitchVodDownloaderSharp/TwitchAPI/APIData.cs
using System;
using System.Collections.Generic;
using System.Runtime.Serialization;

namespace TwitchVodDownloaderSharp.TwitchAPI
{
    //Data Types sent out by API for the downloader to use
    [DataContract]
    public class VideoList
    {
        [DataMember]
        public int _total;
        [DataMember]
        public IList<VideoData> videos;
        [DataMember]
        public Dictionary<string, string> _links;
    }
    [DataContract]
    public class VideoData
    {
        [DataMember]
        public string title;
        [DataMember]
        public string description;
        [DataMember]
        public string broadcast_id;
        [DataMember]
        public string broadcast_type;
        [DataMember]
        public string status;
        [DataMember]
        public string tag_list; //?
        [DataMember]
        public int views;
        [DataMember]
        public string created_at;
        [DataMember]
        public string url;
        [DataMember]
        public string _id;
        [DataMember]
        public string recorded_at;
        [DataMember]
        public string game;
        [DataMember] //seconds
        public long length;
        [DataMember]
        public string preview;
        [DataMember]
        public string animated_preview;
        [DataMember]
        public VideoThumbnail[] thumbnails;
        [DataMember] //includes audio_only
        public Dictionary<string, double> fps;
        [DataMember] //Excludes audio_only
        public Dictionary<string, string> resolutions;
        [DataMember]
        public Dictionary<string, string> _links;
        [DataMember]
        public VideoChannel channel;
    }
    [DataContract]
    public class VideoChannel
    {
        [DataMember]
        public string name;
        [DataMember]
        public string display_name;
    }
    [DataContract]
    public class VideoThumbnail
    {
        [DataMember]
        public string url;
        [DataMember]
        public string type;
    }
    //AccessToken for new TS Vods
    public class AccessToken
    {
        public string token;
        public string sig;
    }

    public class Chunk
    {
        public List<string> m3u_params;
        public string url;

        public long start_timestamp_ms;
        public long end_timestamp_ms;

        public Chunk Clone()
        {
            Chunk nC = new Chunk();
            nC.m3u_params = new List<string>(m3u_params);
            nC.url = url;
            nC.start_timestamp_ms = start_timestamp_ms;
            nC.end_timestamp_ms = end_timestamp_ms;

            return nC;
        }
    }
}
